crates/core/component/compact-block/src/component/manager.rs

```diff
@@ -91,6 +91,8 @@ trait Inner: StateWrite {
         let note_payloads = self
             .pending_note_payloads()
             .into_iter()
+            // Filter out notes with the compact block exclusion flag set.
+            .filter(|(_pos, note, _source)| !note.sync_exclude)
```

Comment:
This will cause an SCT divergence, the note commitments need to be included in the compact block regardless of whether the note payload is included.